tests/analysis/test_utils.py

```diff
@@ -0,0 +1,58 @@
+import bigbang.analysis.utils as utils
+
+import unittest
+import yaml as yaml
+
+import os
+
+TEST_FILENAME = os.path.join(
+    os.path.dirname(__file__), "../data/address_header_test_file.yaml"
+)
+
+
+def extract_email(from_field):
+    """
+    Returns an email address from a string.
+    """
+    match = re.search(email_regex, from_field)
+
+    if match is not None:
+        return match[0].lower()
+
+    else:
+        return None
+
+
+def extract_domain(from_field):
+    """
+    Returns the domain of an email address from a string.
+    """
+    match = re.search(email_regex, from_field)
+
+    if match is not None:
+        return re.search(domain_regex, match[0])[1]
+
+    else:
+        return None
+
+
+"""
+class TestExtractEmail(unittest.TestCase):
```

Comment:
Are we going to need this commented-out code later/elsewhere or could we remove it?

Reply:
I see -- I'll remove it and then merge this PR. Thanks!

Reply:
Hmmm. Actually -- this test file as written does not contain any tests.
The uncommented code here is duplicate code of functions in `bigbang.analysis.utils`.
It looks like somebody was going to write tests for those functions, but did not complete the job.

Reply:
oh -- this whole file is some work in progress I should not have checked in. my bad